lib/_emerge/Scheduler.py

```diff
@@ -1390,6 +1393,22 @@ def _merge(self):
 			self._set_max_jobs(1)
 
 		self._add_prefetchers()
+
+		if not self._build_opts.fetchonly:
+			# Run pkg_pretend concurrently with parallel-fetch, and be careful
+			# to respond appropriately to termination, so that we don't start
+			# any new tasks after we've been terminated.
+			try:
+				rval = self._sched_iface.run_until_complete(self._run_pkg_pretend(loop=self._sched_iface))
+			except asyncio.CancelledError:
+				self.terminate()
+				rval = None
+			self._termination_check()
+			if self._terminated_tasks:
+				rval = 128 + signal.SIGINT
+			if rval != os.EX_OK:
+				return rval
```

Comment:
We need to make sure that --keep-going will behave appropriately when it fails here (and this will fix [bug 404157](https://bugs.gentoo.org/404157)).

Reply:
I've fixed it to add failed packages to self._failed_pkgs, which --keep-going uses to drop those packages from the merge list.